### PR TITLE
Performance: remove count() form the condition section of a loop

### PR DESCRIPTION
**Pull request by coderimus**
### Description
In this PR I removed the `count()` method from condition section for some loops. When this method is used in the loop condition it will be executed every iteration which is not so good for process time (performance).

### Fixed Issues (if relevant)
N/A

### Manual testing scenarios
N/A

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Review by ihor-sviziev — APPROVED** [review]

**Comment by ishakhsuvarov** [comment]
Hi @Coderimus 
Please take a look at the unit test failures.

File: app/code/Magento/Backend/Model/Widget/Grid/Parser.php [hunk 1]
```diff
@@ -30,8 +30,9 @@ public function parseExpression($expression)
         $expression = trim($expression);
         foreach ($this->_operations as $operation) {
             $splittedExpr = preg_split('/\\' . $operation . '/', $expression, -1, PREG_SPLIT_DELIM_CAPTURE);
-            if (count($splittedExpr) > 1) {
```
Comment (ishakhsuvarov) [inline app/code/Magento/Backend/Model/Widget/Grid/Parser.php:33]: I assume that old check would not pass if `$splittedExpr` is and array with 1 element, while new one would pass.

**Review by ishakhsuvarov — CHANGES_REQUESTED** [review]

File: app/code/Magento/Backend/Model/Widget/Grid/Parser.php (hunk 1 above)
Comment (coderimus) [reply app/code/Magento/Backend/Model/Widget/Grid/Parser.php:33]: @ishakhsuvarov thank you for your review! You are totally right. Fix uploaded.

**Comment by coderimus** [comment]
@ishakhsuvarov requested changes added. Please, review them when this will be suitable for you :)

**Comment by ishakhsuvarov** [comment]
Thanks @Coderimus 
Will review and proceed with the merge

**Review by ishakhsuvarov — APPROVED** [review]

**Review by ihor-sviziev — APPROVED** [review]

**Pull request merged**